target/linux/mpc85xx/image/p1020.mk

```diff
@@ -1,20 +1,37 @@
-define Device/aerohive_hiveap-330
+define Device/aerohive_hiveap
   DEVICE_VENDOR := Aerohive
-  DEVICE_MODEL := HiveAP-330
   DEVICE_PACKAGES := kmod-tpm-i2c-atmel
   BLOCKSIZE := 128k
   KERNEL := kernel-bin | gzip | uImage gzip
-  KERNEL_SIZE := 8m
   KERNEL_INITRAMFS := copy-file $(KDIR)/vmlinux-initramfs | uImage none
-  IMAGES := fdt.bin sysupgrade.bin
   IMAGE/fdt.bin := append-dtb
+  IMAGES := fdt.bin sysupgrade.bin kernel.bin rootfs.bin
+  IMAGE/kernel.bin := append-kernel
+  IMAGE/rootfs.bin := append-rootfs | pad-rootfs
+endef
+
+define Device/aerohive_hiveap-330
+  $(Device/aerohive_hiveap)
+  DEVICE_MODEL := HiveAP-330
+  KERNEL_SIZE := 8m
   IMAGE/sysupgrade.bin := append-dtb | pad-to 256k | check-size 256k | \
 	append-uImage-fakehdr ramdisk | pad-to 256k | check-size 512k | \
 	append-rootfs | pad-rootfs $$(BLOCKSIZE) | pad-to 41216k | check-size 41216k | \
 	append-kernel | append-metadata
 endef
 TARGET_DEVICES += aerohive_hiveap-330
 
+define Device/aerohive_hiveap-370
+  $(Device/aerohive_hiveap)
+  DEVICE_MODEL := HiveAP-370
+  DEVICE_PACKAGES += kmod-ath10k-ct ath10k-firmware-qca988x-ct nand-utils block-mount
```

Comment:
Blocking myself here: either the device should be built including `kmod-mtd-rw`, or the `patch_aerohive_ap370_uboot` function above MUST provide a message about needing to manually download and install kmod-mtd-rw while in the initramfs.

Reply:
Cleaned up in commit https://github.com/Hurricos/openwrt/commit/ab67e36741f25d9fa2a9a8bc234c586b952db9dc.